src/sentry/integrations/slack/actions/notification.py

```diff
@@ -117,12 +127,81 @@ def send_notification(event: GroupEvent, futures: Sequence[RuleFuture]) -> None:
                     "attachments": json.dumps(attachments),
                 }
 
+            rule = rules[0] if rules else None
+            rule_to_use = self.rule if self.rule else rule
+            rule_id = rule_to_use.id if rule_to_use else None
+            rule_action_uuid = self.data.get("action_uuid", None)
+            if not rule_action_uuid:
+                # We are logging because this should never happen, all actions should have an uuid
+                # We can monitor for this, and create an alert if this ever appears
+                _default_logger.info(
+                    "integration.slack.actions: No action uuid",
+                    extra={
+                        "rule_id": rule_id,
+                        "notification_uuid": notification_uuid,
+                    },
+                )
+
+            rule_fire_history_id = None
+            try:
+                rule_fire_history = RuleFireHistory.objects.get(notification_uuid=notification_uuid)
+                rule_fire_history_id = rule_fire_history.id
+                if not rule_id:
+                    rule_id = rule_fire_history.rule.id
+            except Exception as err:
+                _default_logger.info(
+                    "integration.slack.actions: Failed to get rule fire history", exc_info=err
+                )
+
+            new_notification_message_object = NewIssueAlertNotificationMessage(
+                rule_fire_history_id=rule_fire_history_id, rule_action_uuid=rule_action_uuid
+            )
+
+            # Only try to get the parent notification message if the organization is in the FF
+            # We need to search by rule action uuid and rule id, so only search if they exist
+            if (
+                features.has(
+                    "organizations:slack-thread-issue-alert", event.group.project.organization
+                )
+                and rule_action_uuid
+                and rule_id
+            ):
+                try:
+                    parent_notification_message = self._repository.get_parent_notification_message(
+                        rule_id=rule_id,
+                        group_id=event.group.id,
+                        rule_action_uuid=rule_action_uuid,
+                    )
+                except Exception:
+                    # if there's an error trying to grab a parent notification, don't let that error block this flow
+                    # we already log at the repository layer, no need to log again here
+                    pass
+                else:
+                    if parent_notification_message:
```

Comment:
the else isn't really doing anything much here, might as well just remove it and keep the if at the same level

Reply:
I can clean this up in my next PR